### PR TITLE
Trev8939/configure basemap style parameters

### PR DESCRIPTION
**Pull request by TADraeseke**
## Description
PR to add a new Kotlin sample  "Configure basemap style parameters" in the Maps category.
## Links and Data
Sample issue: `runtime/kotlin/issues/3847`
## What To Review
-  Correct composable patterns
-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files
## How to Test
Run the sample from this branch

### CODE REVIEW

**Review by hud10837 — CHANGES_REQUESTED** [review]
Works great and easy to understand. I have a few points that might make this a little more idiomatic

File: configure-basemap-style-parameters/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+## How to use the sample
+
+This sample showcases the workflow of configuring basemap style parameters by displaying a basemap with labels in different languages and launches with a `Viewpoint` set over Bulgaria, Greece, and Turkey, as they use three different alphabets: Cyrillic, Greek, and Latin, respectively. By default, the `BasemapStyleLanguageStrategy` is set to `Local` which displays all labels in their corresponding local language. This can be changed to `Global`, which displays all labels in English. The `SpecificLanguage` setting sets all labels to a selected language and overrides the `BasemapStyleLanguageStrategy` settings.
+
+Pan and zoom to navigate the map and see how different labels are displayed in these countries depending on the selected `BasemapStyleLanguageStrategy` and `SpecificLanguage`: all English, all Greek, all Bulgarian, all Turkish, or each their own.
+
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply configure-basemap-style-parameters/README.md:16]: I think I'll leave this out for now. The README is above on other platforms and diverging from them is not preferred

**Comment by TADraeseke** [comment]
Thanks very much for that review @hud10837! Ready for re-review

**Review by hud10837 — APPROVED** [review]
👍 Looks good, easy to understand

File: configure-basemap-style-parameters/src/main/java/com/esri/arcgismaps/sample/configurebasemapstyleparameters/screens/MainScreen.kt [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                                        )
+                                    }
+                                    Divider(
+                                        color = Color.LightGray,
+                                        modifier = Modifier
+                                            .padding(8.dp)
+                                            .fillMaxHeight()
+                                            .width(2.dp)
+                                    )
+                                    Column {
```
Comment (shubham7109) [inline configure-basemap-style-parameters/src/main/java/com/esri/arcgismaps/sample/configurebasemapstyleparameters/screens/MainScreen.kt:143]: You could skip using a divider and split the two columns down the middle: 

```suggestion
                                Row(horizontalArrangement = Arrangement.SpaceBetween) {
                                    Column(Modifier.weight(0.5f)) {
                                        // UI for setting the language strategy
                                        LanguageStrategyControls(
                                            languageStrategyOptions = mapViewModel.languageStrategyOptions,
                                            onLanguageStrategyChange = { languageStrategy ->
                                                mapViewModel.updateLanguageStrategy(languageStrategy)
                                            },
                                            languageStrategy = mapViewModel.languageStrategy,
                                            enabled = (mapViewModel.specificLanguage == "None")
                                        )
                                    }

                                    Column(Modifier.weight(0.5f)) {
```

**Review by shubham7109 — COMMENTED** [review]
Instead of disabling the `LanguageStrategyControls` I would suggest to disable the `SpecificLanguageControls` if "Local" strategy is used (except None). Or maybe auto-select "Global" if the user decides to choose a specific language. 

Using this, we can avoid the scenario if the user was to choose "Local" strategy and choose a specific language. Which doesn't look right to in my opinion.

Screenshot 2024-05-28 at 5 58 51 PM


Sample looks good, added some general feedback comments. Nice work @TADraeseke 👍

**Comment by puneet-pdx** [comment]
@TADraeseke I can review/approve this when you are done addressing @shubham7109 comments.

**Comment by TADraeseke** [comment]
> Instead of disabling the `LanguageStrategyControls` I would suggest to disable the `SpecificLanguageControls` if "Local" strategy is used (except None). Or maybe auto-select "Global" if the user decides to choose a specific language.
> 
> Using this, we can avoid the scenario if the user was to choose "Local" strategy and choose a specific language. Which doesn't look right to in my opinion.
> 
> Screenshot 2024-05-28 at 5 58 51 PM
> 
> Sample looks good, added some general feedback comments. Nice work @TADraeseke 👍

Thanks @shubham7109 -- disabling the language strategy controls, as seen in this sample, was set out in the common samples design. https://devtopia.esri.com/runtime/common-samples/tree/main/designs/ConfigureBasemapStyleParameters#how-it-works

Not sure if we want to go back and suggest inverting that UI behaviour at this stage? Could be a good one for the samples council I guess!

**Comment by TADraeseke** [comment]
> @TADraeseke I can help review/approve this when you are done addressing @shubham7109 comments.

Thanks @puneet-pdx! Shubham is actually the second reviewer on this one already, but I've just opened another PR for a new sample that I'd be grateful for your review on if you're keen! https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/204

**Comment by puneet-pdx** [comment]
@TADraeseke I was trying to help finish this review in this sprint as @shubham7109 is out for the next week. If you want @shubham7109 to approve this then I will hold back.

**Review by puneet-pdx — APPROVED** [review]
@TADraeseke some minor comments. Looks good overall.

**Comment by TADraeseke** [comment]
Thanks @puneet-pdx! Made those fixes. Merging!

**Pull request merged**